src/autoqasm/types/types.py

```diff
@@ -104,8 +104,12 @@ def __init__(self, *args, annotations: str | Iterable[str] | None = None, **kwar
             raise errors.InvalidArrayDeclaration(
                 "Arrays may only be declared at the root scope of an AutoQASM main function."
             )
+
+        if not args:
+            raise errors.InvalidArrayDeclaration("init_expression must at least be provided.")
```

Comment:
This is nice, thank you! If we're going enforce that `init_expression` exists, I think we should explicitly add it to the `__init__` function signature here - can you do that as well?

Reply:
Thanks, that's a great idea! Have made the change.